src/Illuminate/Http/Request.php

```diff
@@ -193,14 +193,23 @@ public function is()
     }
 
     /**
-     * Check if the route name matches the given string.
+     * Determine if the route name matches a pattern.
      *
-     * @param  string  $name
      * @return bool
      */
-    public function routeIs($name)
+    public function routeIs()
     {
-        return $this->route() && $this->route()->named($name);
+        if ((! $route = $this->route()) || ! $routeName = $route->getName()) {
```

Comment:
Without that parenthesis around the assignment of `$route`, the IDE is showing the next use of `$route` as an error on the assignment of `$routeName`